discovery/gossiper.go

```diff
@@ -1072,39 +1151,23 @@ func (d *AuthenticatedGossiper) networkHandler() {
 				continue
 			}
 
-			// For the set of peers that have an active gossip
-			// syncers, we'll collect their pubkeys so we can avoid
-			// sending them the full message blast below.
-			syncerPeers := d.syncMgr.GossipSyncers()
-
-			log.Infof("Broadcasting batch of %v new announcements",
+			subBatchSize := calculateSubBatchSize(
+				d.cfg.TrickleDelay,
+				d.cfg.SubBatchDelaySecond,
 				len(announcementBatch))
 
-			// We'll first attempt to filter out this new message
-			// for all peers that have active gossip syncers
-			// active.
-			for _, syncer := range syncerPeers {
-				syncer.FilterGossipMsgs(announcementBatch...)
-			}
+			splitAnnouncementBatch := splitAnnouncementBatches(
+				subBatchSize,
+				d.cfg.MinimumBatchSize,
+				announcementBatch)
 
-			// Next, If we have new things to announce then
-			// broadcast them to all our immediately connected
-			// peers.
-			for _, msgChunk := range announcementBatch {
-				// With the syncers taken care of, we'll merge
-				// the sender map with the set of syncers, so
-				// we don't send out duplicate messages.
-				msgChunk.mergeSyncerMap(syncerPeers)
-
-				err := d.cfg.Broadcast(
-					msgChunk.senders, msgChunk.msg,
-				)
-				if err != nil {
-					log.Errorf("unable to send batch "+
-						"announcements: %v", err)
-					continue
+			go func() {
```

Comment:
we'll want to make sure that this goroutine is waitgrouped, so that `Stop` will block until it exits. to do so, we use the following pattern:
```golang
d.wg.Add(1)
go func() {
        defer d.wg.Done()
```